### PR TITLE
Added Cart page

### PR DESCRIPTION
**Pull request by jeetsdev**
Added cart page and functionalities.
- User can add to cart
- User can change item quantity 
- User can remove it from the cart 
- Price
- Discount price 
- Final Price 

[Preview here](https://deploy-preview-3--thriving-gumption-494d22.netlify.app/)

### CODE REVIEW

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -33,9 +26,11 @@ export const Header = () => {
                 </div>
                 {/* Button section here*/}
                 <div className="nav__button-sec center__flex">
-                    <Link to="/" className="btns btn__primary border__rad-4px">Login</Link>
-                    <HeaderIcon iconName={"fa-heart"} />
-                    <HeaderIcon iconName={"fa-shopping-bag"} />
+                    <Link to="/products" className="btns btn__primary margin__lr-8px border__rad-4px">Shop</Link>
+                    <i className="fas fa-sign-in-alt margin-8px"></i>
+                    {/* <i className="fas fa-sign-out-alt"></i> */}
```
Comment (znjs) [inline src/components/Header/Header.jsx]: can remove the commented code

File: src/contexts/cart-context.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export const CartProvider = ({ children }) => {
+  const [cartState, cartDispatch] = useReducer(cartReducer, {
+    cartItem: [],
+    loader: false,
+    totalPrice: 0,
+  });
+
+  useEffect(() => {
+    
+  })
```
Comment (znjs) [inline src/contexts/cart-context.js]: can remove the unused `useEffect`

File: src/App.js [hunk 3]
```diff
@@ -4,7 +4,6 @@ import { NavigationRoutes } from "./routes/NavigationRoutes";
 function App() {
   return (
     <div className="App">
-      {/* {console.log("rendering")} */}
```
Comment (ragini07) [inline src/App.js:7]: remove all console logs

File: src/components/Cards/ProductCard.jsx [hunk 4]
```diff
@@ -23,11 +28,12 @@ export const ProductCard = ({ product }) => {
 
                 {/*Card overlay section here  */}
                 <div className="card__upper-overlay">
-                    <div className="upper__overlay-icon">
-                        <p className="shopping__cart-icon center__flex"> + </p>
+                    <div className="upper__overlay-icon" onClick={() => cartDispatch({ type: ADD_TO_CART,payload:product})}>
+                        {/* <p className="shopping__cart-icon center__flex"> + </p>
+                        <p className="shopping__cart-icon center__flex"> - </p> */}
```
Comment (ragini07) [inline src/components/Cards/ProductCard.jsx]: remove commented code if not needed

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (ragini07) [inline src/components/Header/Header.jsx]: same here

**Review by znjs — APPROVED** [review]
- Good folder structure
- Nice that you used constants instead of strings
- Easy to understand the code (proper comments mentioned)

**Review by ragini07 — COMMENTED** [review]
-Liked that you used barrel export
-Good file structure
-Suggestion : Can interchange the + and - icon in cart page as it looks odd to see the + icon on left side as this not what we usually see on ecommerce sites.
-Rest looks good

**Review by ragini07 — APPROVED** [review]

**Pull request merged**